osconfig_tests/test_config/project_config.go

```diff
@@ -52,24 +52,45 @@ func GetProject(projectID string, testZones map[string]int) *Project {
 
 // GetZone gets a random zone that still has capacity.
 func (p *Project) GetZone() string {
-	p.mux.Lock()
-	defer p.mux.Unlock()
+	for {
+		p.mux.Lock()
+
+		zc := len(p.zoneIndices)
+		if zc == 0 {
+			p.mux.Unlock()
+			time.Sleep(10 * time.Second)
```

Comment:
should we have a upper bound for the amount of time it should wait before just bailing out the execution?

Reply:
Maybe, we do have a timeout on the test though so I can look into adding this later

Reply:
Actually this was simple to implement